### ISSUE TITLE
Numeric.LinearAlgebra.Static: show produces scrambled output for vectors

### ISSUE THREAD

**Issue by martin-neuhaeusser** [haskell-numerics/hmatrix#289]
The output of show for dependently typed vectors is scrambled on my system.
The sample program
```haskell
{-# LANGUAGE DataKinds #-}

module Main where

import qualified Numeric.LinearAlgebra.Static as LA

main :: IO()
main =
  let v = (LA.linspace (0, 10) :: LA.R 5)
  in
    do
      putStrLn ("Extracted vector shows as: " ++ show (LA.extract v))
      putStrLn ("Static vector shows as   : " ++ show v)
```
produces the output

```
Extracted vector shows as: [0.0,2.5,5.0,7.5,10.0]
Static vector shows as   : (vector,5.0,7.5,10.0] :: R 5)
```
where the first line displays as I would expect; however, when showing the dependently typed vector directly (see second line), the output is somehow scrambled and does not display the first two dimensions.

**Comment by idontgetoutmuch** [comment]
Thanks for the report :) I think I knew about this. Perhaps you could submit a PR?

**Comment by martin-neuhaeusser** [comment]
This bug was reported and fixed before I opened the issue. 
The reason I stumbled across it was that the version of hmatrix on hackage was lacking behind the github versions. The bug is resolved on hackage since hmatrix 0.20.0.0.
